### ISSUE TITLE
(node:52739) Warning: Accessing non-existent property 'cat' of module exports inside circular dependency

### ISSUE THREAD

**Issue by namenotrequired** [SublimeLinter/SublimeLinter#1783]
SublimeLinter does show me errors, but every time I open or save a file, it gives a bunch of errors that seem to be from nodejs. See print below.

The only linter I have installed is SublimeLinter-eslint. My SublimeLinter settings file is empty.

I uninstalled it and reinstalled it and the problem continues.

I uninstalled all my other packages and the problem continues.

It seems to be a Node issue, so in case it's relevant, I checked the node version on my machine. `$ node -v` gave `v14.15.0`

## Errors

<img width="791" alt="Screen Shot 2021-03-03 at 11 18 00" src="https://user-images.githubusercontent.com/3224761/109790934-2fe93400-7c12-11eb-9b2d-27f287867e87.png">

## Logs

```

(node:52858) Warning: Accessing non-existent property 'cat' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:52858) Warning: Accessing non-existent property 'cd' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'chmod' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'cp' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'dirs' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'pushd' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'popd' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'echo' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'tempdir' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'pwd' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'exec' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'ls' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'find' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'grep' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'head' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'ln' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'mkdir' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'rm' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'mv' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'sed' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'set' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'sort' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'tail' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'test' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'to' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'toEnd' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'touch' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'uniq' of module exports inside circular dependency
(node:52858) Warning: Accessing non-existent property 'which' of module exports inside circular dependency
```

**Comment by kaste** [comment]
Hm, that's typical a version mismatch between eslint and the used node.  Simplest case is you updated node and now eslint is outdated etc.  If you have a sophisticated setup/environment (say nvm or something like that) it could get more complicated to resolve, then maybe the environment of the Sublime Text process differs from the Terminal experience...

**Comment by namenotrequired** [comment]
Thanks kaste! My setup is very simple. Is there a way I can update the node version of the plugin? Reinstalling the plugin didn't solve the issue

**Comment by kaste** [comment]
Oh, that's a *very* async communication here.  Anyway, maybe you have a system node installed which gets picked up here.  

If *eslint* works on the terminal, try to close all Sublime, then start it again from the "correct" working dir from the terminal using `subl .`.  This makes Sublime inherit the process environment from the Terminal.

**Comment by namenotrequired** [comment]
Thanks. I just got eslint working on the terminal, and restarted sublime, but the problem persists and `$ subl .` gives me `-bash: subl: command not found`

**Comment by kaste** [comment]
For the *subl* command refer https://www.sublimetext.com/docs/command_line.html

**Comment by kaste** [comment]
Closing as there is no actionable issue in sight.  Comments are still open of course.  👍

**Comment by namenotrequired** [comment]
For anyone else who has this in the future, updating node did nothing but updating my eslint-related packages listed in `package.json` of each project solved this for me. Thanks @kaste for the help.

**Comment by j13ag0** [comment]
Thanks @namenotrequired for that.
In my case, getting rid of this was the solution:
`"eslint-rallycoding": "^1.2.0",`
